### PR TITLE
Migrate MiniCPMVImageInputs to TensorSchema

### PR DESCRIPTION
**Pull request by bbeckca**
## Purpose
This PR migrates MiniCPMVImageInputs from a TypedDict-based definition to a structured TensorSchema model with runtime shape validation. This brings it in line with recent changes to Phi3VImagePixelInputs, and is part of a broader effort to improve input contract enforcement and debug-ability across multi-modal models.

## Test Plan

Confirm validation works via standalone tests in tests/standalone_test/test_tensor_schema.py and rely on CI to check integration.

## Test Result

```
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_valid_tensor PASSED                                                                                                                                      [  6%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_optional_fields PASSED                                                                                                                                   [ 12%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_constant_dim_failure PASSED                                                                                                                              [ 18%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_symbolic_dim_mismatch PASSED                                                                                                                             [ 25%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_list_tensor_valid PASSED                                                                                                                                 [ 31%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_variable_patch_counts_valid PASSED                                                                                                                       [ 37%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_tuple_tensor_valid PASSED                                                                                                                                [ 43%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_inconsistent_shapes_in_list PASSED                                                                                                                       [ 50%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_empty_list PASSED                                                                                                                                        [ 56%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_validation_disabled_skips_shape_check PASSED                                                                                                             [ 62%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_with_valid_resolve_binding_dims PASSED                                                                                                                   [ 68%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_with_invalid_resolve_binding_dims PASSED                                                                                                                 [ 75%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_with_list_of_symbolic_dim PASSED                                                                                                                         [ 81%]
tests/standalone_tests/test_tensor_schema.py::test_tensor_schema_with_list_of_symbolic_dim_mismatch_in_length PASSED                                                                                                      [ 87%]
tests/standalone_tests/test_tensor_schema.py::test_valid_tensor_schema_with_static_last_dim PASSED                                                                                                                        [ 93%]
tests/standalone_tests/test_tensor_schema.py::test_invalid_tensor_schema_with_static_last_dim PASSED                                                                                                                      [100%]
```

### CODE REVIEW

File: tests/standalone_tests/test_tensor_schema.py [hunk 1]
```diff
@@ -129,23 +129,31 @@ def test_tensor_schema_with_invalid_resolve_binding_dims():
...
+    # tgt_sizes = (3, 2) → bns = 3 (inconsistent)
+    tgt_sizes = torch.randint(0, 256, (3, 2))
+    num_slices = torch.tensor([2, 2])
+
+    with pytest.raises(ValueError, match="expected 'bns'=4, got 3"):
+        MiniCPMVImagePixelInputs(
+            pixel_values=pixel_values,
+            tgt_sizes=tgt_sizes,
+            num_slices=num_slices,
         )
```
Comment (bbeckca) [inline tests/standalone_tests/test_tensor_schema.py]: @DarkLight1337 @Isotr0py Heads up, updating tests that are no longer valid following fix for `FuyuImagePatchInputs`. #21709

File: vllm/model_executor/models/minicpmv.py [hunk 2]
```diff
@@ -72,36 +73,47 @@
...
-    """
-    Shape: `(batch_size * num_images, num_slices, hidden_size)`
-
-    `hidden_size` must match the hidden size of language model backbone.
-    instead of a batched tensor.
-    """
+    image_embeds: Annotated[
+        Union[torch.Tensor, list[torch.Tensor]],
+        TensorShape("bn", "ns", "hs"),
+    ]
```
[the remark this answers is not in the document]
Comment (bbeckca) [reply vllm/model_executor/models/minicpmv.py:118]: This wasn't enforced previously, so I haven't added any validations. Feel free to let me know if we should make `ns` dynamic dim.

File: tests/standalone_tests/test_tensor_schema.py (hunk 1 above)
Comment (bbeckca) [reply tests/standalone_tests/test_tensor_schema.py]: @DarkLight1337 @Isotr0py Just noticed this overlaps with a fix I recently published (#22366).  Let me know if we'd prefer to just land the updated tests as part of this migration and abandon that one.

File: vllm/model_executor/models/minicpmv.py [hunk 3]
```diff
@@ -72,36 +73,47 @@
...
 
-    tgt_sizes: torch.Tensor
+    type: Literal["pixel_values"] = "pixel_values"
+
+    # Note that the image size may vary, so we pass it as a list instead of a
+    # batched tensor.
+    pixel_values: Annotated[
+        list[torch.Tensor],
+        TensorShape("bns", 3, "h", "w"),
+    ]
```
[the remark this answers is not in the document]
Comment (bbeckca) [reply vllm/model_executor/models/minicpmv.py]: I think we're fine to leave as is since this wasn't enforced and values aren't used in any symbolic cross field checks.

**Review by Isotr0py — APPROVED** [review]
